src/script/script.cpp

```diff
@@ -255,6 +256,60 @@ bool CScript::IsPayToScriptHash() const
             (*this)[22] == OP_EQUAL);
 }
 
+bool CScript::IsRedeemScriptReveal(CScript scriptpubkey) const{
+    CScript check_spk, redeemScript = scriptpubkey;
+    if (
+        // these magic numbers correspond to a typical(as of May 2021) atomicdex HLTC
+        redeemScript[0] == OP_RETURN &&
+        redeemScript[3] == OP_IF &&
+        redeemScript[9] == OP_NOP2 &&
+        redeemScript[10] == OP_DROP &&
+        redeemScript[45] == OP_CHECKSIG &&
+        redeemScript[46] == OP_ELSE &&
+        redeemScript[47] == OP_SIZE &&
+        redeemScript[48] == 0x01 &&
+        redeemScript[49] == 0x20 &&
+        redeemScript[50] == OP_EQUALVERIFY &&
+        redeemScript[51] == OP_HASH160 &&
+        redeemScript[73] == OP_EQUALVERIFY &&
+        redeemScript[108] == OP_CHECKSIG &&
+        redeemScript[109] == OP_ENDIF &&
+        redeemScript.size() == 110
+    ) {
+        // Drop the OP_RETURN and OP_PUSHDATA1 + byte
+        redeemScript.erase(redeemScript.begin(),redeemScript.begin()+3 );
+        check_spk << OP_HASH160 << ToByteVector(CScriptID(redeemScript)) << OP_EQUAL;
+        return (check_spk == (*this));
+    }
+
+    if (
+        // these magic numbers correspond to:
+        // 16_arbitrary_bytes OP_DROP 
+        // OP_IF 32bit_locktime OP_NOP2 OP_DROP pubkey OP_CHECKSIG 
+        // OP_ELSE pubkey OP_CHECKSIG OP_ENDIF
+        // as provided by artem 
+        redeemScript[0] == OP_RETURN &&
+        redeemScript[1] == 0x4c && // PUSH_BYTES
+        redeemScript[2] == 0x62 && // BYTES
+        redeemScript[20] == OP_DROP &&
+        redeemScript[21] == OP_IF &&
+        redeemScript[22] == 0x04 && // 32bit locktime
+        redeemScript[27] == OP_NOP2 &&
+        redeemScript[28] == OP_DROP &&
+        redeemScript[63] == OP_CHECKSIG &&
+        redeemScript[64] == OP_ELSE &&
+        redeemScript[99] == OP_CHECKSIG &&
+        redeemScript[100] == OP_ENDIF &&
+        redeemScript.size() == 101
+    ) {
+        // Drop the OP_RETURN and OP_PUSHDATA1 + byte
+        redeemScript.erase(redeemScript.begin(),redeemScript.begin()+3 );
+        check_spk << OP_HASH160 << ToByteVector(CScriptID(redeemScript)) << OP_EQUAL;
+        return (check_spk == (*this));
+    }
+    return(0);
```

Comment:
I don't have other comments except the only one. It's just unclear to return an integer instead of a boolean